packages/blockchain-link-utils/src/ripple.ts

```diff
@@ -1,62 +1,69 @@
-import type { Transaction } from '@trezor/blockchain-link-types';
+import type { AccountTxTransaction, ServerInfoResponse } from 'xrpl';
 
-type RippleTransactionMetadata = {
-    AffectedNodes: any[];
-    DeliveredAmount?: any[];
-    TransactionIndex: number;
-    TransactionResult: string;
-    delivered_amount?: string;
-};
+import type { Transaction } from '@trezor/blockchain-link-types';
 
-// export const transformServerInfo = (payload: GetServerInfoResponse) => {
-export const transformServerInfo = (payload: any) => ({
+export const transformServerInfo = (payload: ServerInfoResponse) => ({
     name: 'Ripple',
     shortcut: 'xrp',
     network: 'xrp',
     testnet: false,
-    version: payload.buildVersion,
+    version: payload.result.info.build_version,
     decimals: 6,
-    blockHeight: payload.validatedLedger.ledgerVersion,
-    blockHash: payload.validatedLedger.hash,
+    blockHeight: payload.result.info.validated_ledger?.seq,
+    blockHash: payload.result.info.validated_ledger?.hash,
 });
 
-// https://bitcoin.stackexchange.com/questions/23061/ripple-ledger-time-format/23065#23065
+// XRPL timestamps are based on the Ripple Epoch (2000-01-01T00:00:00Z).
+// To convert to a standard Unix timestamp, add this offset.
+// https://xrpl.org/docs/references/protocol/data-types/basic-data-types#specifying-time
 const BLOCKTIME_OFFSET = 946684800;
 
+const getUnixTimestamp = (xrplTimestamp?: number): number => {
+    if (!xrplTimestamp || xrplTimestamp <= 0) {
+        return 0;
+    }
+
+    return xrplTimestamp + BLOCKTIME_OFFSET;
+};
+
 export const transformTransaction = (
-    tx: any,
-    meta: RippleTransactionMetadata | null,
+    hash: string | undefined,
+    tx_json: NonNullable<AccountTxTransaction['tx_json']>,
+    meta: AccountTxTransaction['meta'] | undefined,
     descriptor?: string,
 ): Transaction => {
-    const blockTime =
```

Comment:
I would first check if TransactionType is `Payment` and if yes, you do not need to do those `@ts-expect-error`s

I think all other txs are unknown for us (would be good to check it)